tests/vm_delete_protection_test.go

```diff
@@ -0,0 +1,78 @@
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/rand"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"kubevirt.io/ssp-operator/tests/env"
+)
+
+const deleteProtectionLabel = "kubevirt.io/vm-delete-protection"
+
+var _ = Describe("VM delete protection", func() {
+
+	var vm *kubevirtv1.VirtualMachine
+
+	BeforeEach(func() {
+		waitUntilDeployed()
+	})
+
+	AfterEach(func() {
+		err := apiClient.Get(ctx, client.ObjectKeyFromObject(vm), vm)
+		Expect(err).To(Or(Not(HaveOccurred()), MatchError(errors.IsNotFound, "VM not found")))
+
+		if err == nil {
+			Eventually(func() error {
+				if err := apiClient.Get(ctx, client.ObjectKeyFromObject(vm), vm); err != nil {
+					return err
+				}
+
+				vm.Labels[deleteProtectionLabel] = "False"
+				return apiClient.Update(ctx, vm)
+			}, env.ShortTimeout(), time.Second).Should(Succeed())
+
+			Expect(apiClient.Delete(ctx, vm)).To(Succeed())
+			waitForDeletion(client.ObjectKeyFromObject(vm), &kubevirtv1.VirtualMachine{})
+		}
+	})
+
+	It("should not allow to delete a VM if the protection is enabled", func() {
+		vm = createVMWithDeleteProtection(map[string]string{deleteProtectionLabel: "True"})
+
+		Expect(apiClient.Delete(ctx, vm)).To(
+			MatchError(ContainSubstring("VirtualMachine %v cannot be deleted, remove delete protection", vm.Name)))
+	})
+
+	It("should be able to delete a VM if the protection is disabled", func() {
+		vm = createVMWithDeleteProtection(map[string]string{deleteProtectionLabel: "False"})
+
+		Expect(apiClient.Delete(ctx, vm)).To(Succeed())
+		waitForDeletion(client.ObjectKeyFromObject(vm), &kubevirtv1.VirtualMachine{})
+	})
+
+	It("should be able to delete a VM if the VM does not have any label", func() {
+		vm = createVMWithDeleteProtection(nil)
```

Comment:
The name `createVMWithDeleteProtection` no longer holds true. Can you make this func take a bool argument and extract the basic VM generation logic to another helper and use it here and in `createVMWithDeleteProtection`?

Reply:
Done.